docs/buildsystem.rst

```diff
@@ -785,6 +785,36 @@ A typical use for this tool is creating static libraries.
    preferable in future to correctly update/delete/create the archive file
    as required.
 
+Shared Library Tool
+------------
+
+**Identifier**: *shared-library*
+
+A tool used to create a shared library (``.so``, ``.dll``)
+
+.. list-table::
+   :header-rows: 1
+   :widths: 20 80
+
+   * - Name
+     - Description
+
+   * - swiftExecutable
+     - A string indicating the path to a ``swiftc`` compiler that will be used
+       to link Swift code.
+
+   * - clangExecutable
+     - A string indicating the path to a ``clang`` compiler that will be used
+       to link C code.
+
+   * - clang++Executable
```

Comment:
It feels weird having "+" in an identifier, can you please use something like "clangPlusPlusExecutable" instead?